ogc/src/main/scala/geotrellis/server/ogc/OgcSource.scala

```diff
@@ -212,16 +218,23 @@ case class MapAlgebraSourceMetadata(
 case class MapAlgebraSource(
   name: String,
   title: String,
-  sources: Map[String, RasterSource],
+  ogcSources: Map[String, RasterOgcSource],
   algebra: Expression,
   defaultStyle: Option[String],
   styles: List[OgcStyle],
   resampleMethod: ResampleMethod,
   overviewStrategy: OverviewStrategy,
-  timeMetadataKey: Option[String]
+  timeFormat: OgcTimeFormat
 ) extends OgcSource {
+  // each of the underlying ogcSources uses it's own timeMetadataKey
+  val timeMetadataKey: Option[String] = None
+
+  lazy val sources: Map[String, RasterSource]    = ogcSources.mapValues(_.source)
+  lazy val sourcesList: List[RasterSource]       = sources.values.toList
+  lazy val ogcSourcesList: List[RasterOgcSource] = ogcSources.values.toList
+
   def extentIn(crs: CRS): Extent = {
-    val reprojectedSources: NEL[RasterSource] = NEL.fromListUnsafe(sources.values.map(_.reproject(crs)).toList)
+    val reprojectedSources: NEL[RasterSource] = NEL.fromListUnsafe(sourcesList.map(_.reproject(crs)))
```

Comment:
Why is this `fromListUnsafe` call safe? Can we bubble up the non-emptiness requirement to avoid this? (again I know it's old but I'm thinking about it)

Reply:
@jisantuc I guess that is possible, but that would be a refactor that deserves a separate PR. Instead of MAPs all over the places it would mean to use NonEmptyMaps... and these NonEmptyMaps should be some how collected from potentially empty lists. 

That's an unfrotunate the MosaicRasterSource caused inconsistenies in the RasterSources API.

Reply:
@jisantuc https://github.com/geotrellis/geotrellis-server/issues/353